pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java

```diff
@@ -63,20 +63,21 @@
 
   @Override
   public void init(PinotConfiguration config) {
-    LOGGER.info("Configs are: {}, {}",
-            PROJECT_ID,
-            config.getProperty(PROJECT_ID));
+    Credentials credentials;
 
-    checkArgument(!isNullOrEmpty(config.getProperty(PROJECT_ID)));
-    checkArgument(!isNullOrEmpty(config.getProperty(GCP_KEY)));
-    String projectId = config.getProperty(PROJECT_ID);
-    String gcpKey = config.getProperty(GCP_KEY);
     try {
-      storage = StorageOptions.newBuilder()
-          .setProjectId(projectId)
-          .setCredentials(GoogleCredentials.fromStream(Files.newInputStream(Paths.get(gcpKey))))
-          .build()
-          .getService();
+      StorageOptions.Builder storageBuilder = StorageOptions.newBuilder();
+      if (!isNullOrEmpty(config.getProperty(PROJECT_ID)) && !isNullOrEmpty(config.getProperty(GCP_KEY))) {
+        LOGGER.info("Configs are: {}, {}", PROJECT_ID, config.getProperty(PROJECT_ID));
+        String projectId = config.getProperty(PROJECT_ID);
+        String gcpKey = config.getProperty(GCP_KEY);
+        storageBuilder.setProjectId(projectId);
```

Comment:
Just curious about this `storageBuilder.setProjectId(...)`

Shall we allow users to just specify `PROJECT_ID` and use default credential for `GCP_KEY`

Reply:
Also, shall we check the presence of `PROJECT_ID` always? Or it's part from the application default?

Reply:
PROJECT_ID is a part of service account. 
I think in passing credentials manually way, we dont need PROOECT_ID also. 
Ref doc: https://cloud.google.com/docs/authentication/production#linux-or-macos

Reply:
cool, thanks for the clarification!